install.sh

```diff
@@ -0,0 +1,567 @@
+# extension. Note: Most shells limit `local` to 1 var per line, contra bash.
+
+if [ "$KSH_VERSION" = 'Version JM 93t+ 2010-03-05' ]; then
+    # The version of ksh93 that ships with many illumos systems does not
+    # support the "local" extension.  Print a message rather than fail in
+    # subtle ways later on:
+    echo 'temporal does not work with this ksh93 version; please try bash!' >&2
+    exit 1
+fi
+
+set -u
+
+#XXX: If you change anything here, please make the same changes in setup_mode.rs
+usage() {
+    cat 1>&2 <<EOF
+temporal.sh 0.1.0
+The installer for temporal
+
+USAGE:
+    sh install.sh
+EOF
+}
+
+main() {
+    downloader --check
+    need_cmd uname
+    need_cmd mktemp
+    need_cmd chmod
+    need_cmd mkdir
+    need_cmd rm
+    need_cmd rmdir
+    need_cmd tar
+
+    get_architecture || return 1
+    local _arch="$RETVAL"
+    assert_nz "$_arch" "arch"
+
+    get_platform || return 1
+    local _platform="$RETVAL"
+    assert_nz "$_platform" "platform"
+
+    local _ansi_escapes_are_valid=false
+    if [ -t 2 ]; then
+        if [ "${TERM+set}" = 'set' ]; then
+            case "$TERM" in
+            xterm* | rxvt* | urxvt* | linux* | vt*)
+                _ansi_escapes_are_valid=true
+                ;;
+            esac
+        fi
+    fi
+
+    # check if we have to use /dev/tty to prompt the user
+    local need_tty=yes
+    for arg in "$@"; do
+        case "$arg" in
+        --help)
+            usage
+            exit 0
+            ;;
+        *)
+            OPTIND=1
+            if [ "${arg%%--*}" = "" ]; then
+                # Long option (other than --help);
+                # don't attempt to interpret it.
+                continue
+            fi
+            while getopts :hy sub_arg "$arg"; do
+                case "$sub_arg" in
+                h)
+                    usage
+                    exit 0
+                    ;;
+                y)
+                    # user wants to skip the prompt --
+                    # we don't need /dev/tty
+                    need_tty=no
+                    ;;
+                *) ;;
+
+                esac
+            done
+            ;;
+        esac
+    done
+
+    if $_ansi_escapes_are_valid; then
+        printf "\33[1minfo:\33[0m downloading Temporal CLI\n" 1>&2
+    else
+        printf '%s\n' 'info: downloading Temporal CLI' 1>&2
+    fi
+
+    # TODO replace with
+    # local _url="https://temporal.download/temporalite/archive/latest?platform=${_platform}&arch=${_arch}"
+    local _url="https://temporal.download/assets/temporalio/cli/releases/download/v0.2.0/cli_0.2.0_${_platform}_${_arch}.tar.gz"
+
+    local _ext="tar.gz"
+    case "$_arch" in
+    *windows*)
+        _ext=".zip"
+        ;;
+    esac
+
+    local _dir="$(ensure get_install_dir)"
+    ensure mkdir -p "$_dir"
+
+    local _archive="${_dir}/temporal_cli_latest${_ext}"
+    ensure downloader "$_url" "$_archive" "$_arch"
+    ensure unzip "$_archive" "$_dir"
```

Comment:
Does this work on windows?

Reply:
no i haven't tested on Windows and it's not going to currently work since unzip() doesn't explicitly call `zip`. I want to test Windows via the Github Actions test, likely in a new PR

Reply:
Okay, let's make sure to track this.